packages/compass-crud/src/components/document-list.tsx

```diff
@@ -260,15 +268,25 @@ class DocumentList extends React.Component<DocumentListProps> {
 
     return (
       <div className="document-list-zero-state">
+        <ConfirmationModal
+          title="Import data to the collection"
+          open={this.state.showImportConfirmationModal}
+          onConfirm={this.onConfirmImportDataClicked.bind(this)}
+          onCancel={this.onShowImportConfirmationModalToggled.bind(this)}
+          buttonText="Select file..."
+          data-testid="import-new-data-confirm-modal"
+        >
+          It only takes a few seconds to import data from a JSON or CSV file.
+        </ConfirmationModal>
         <EmptyContent
           icon={DocumentIcon}
           title="This collection has no data"
           subTitle="It only takes a few seconds to import data from a JSON or CSV file."
           callToAction={
             <Button
               disabled={!this.props.isEditable}
-              onClick={this.props.openImportFileDialog}
-              data-testid="import-data-button"
+              onClick={this.onShowImportConfirmationModalToggled.bind(this)}
```

Comment:
Instead of adding confirmation state and logic in the component, you can use `showConfirmation` method that @mabaasit added recently in https://github.com/mongodb-js/compass/pull/4079 and replace all this code with something like

 ```suggestion
              onClick={() => {
                showConfirmation(...).then(confirmed => {
                  if (confirmed) {
                    this.props.openImportFileDialog()
                  }
                })
              }}
```